src/update-process-env.js

```diff
@@ -0,0 +1,49 @@
+/** @babel */
+
+import {spawnSync} from 'child_process'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set(['NODE_ENV', 'NODE_PATH'])
+
+export default function updateProcessEnv (launchEnv) {
+  let envToAssign
+  if (launchEnv.PWD) {
+    envToAssign = launchEnv
+  } else {
+    if (process.platform === 'darwin') {
+      envToAssign = getEnvFromShell()
+    }
+  }
+
+  if (envToAssign) {
+    for (let key in process.env) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        delete process.env[key]
+      }
+    }
+
+    for (let key in envToAssign) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        process.env[key] = envToAssign[key]
+      }
+    }
+  }
+}
+
+function getEnvFromShell () {
```

Comment:
I think your last statements creating false dichotomies. It is possible to make Atom a great text editor without taking on dependencies in all areas where you could depend on something. And having code in this specific area does not make Atom "bloatware".

If you believe the patching behavior should differ in some way you should file an issue substantiating the ways in which it should differ. The PATH patching you have to do for linters is a good example of a potential issue you could open. I don't currently understand *why* you need to do this, so opening an issue would enhance our shared understanding of your needs.

In general, further discussion of this in this line comment trail seems counter productive as the code has changed materially.